api/src/main/java/org/openmrs/api/OrderService.java

```diff
@@ -785,4 +785,13 @@ public Order discontinueOrder(Order orderToDiscontinue, String reasonNonCoded, D
 	 */
 	@Authorized({ PrivilegeConstants.EDIT_ORDERS, PrivilegeConstants.ADD_ORDERS })
 	public OrderGroup saveOrderGroup(OrderGroup orderGroup) throws APIException;
+
+	/**
+	 * Given a patient or an encounter or both, fetches List of OrderGroups.
+	 * @param patient the patient.
+	 * @param encounter the encounter.
+	 * @return List of OrderGroup objects
+	 */
+	@Authorized(PrivilegeConstants.GET_ORDERS)
+	public List<OrderGroup> getListOfOrderGroupsByPatientAndEncounter(Patient patient, Encounter encounter);
```

Comment:
Patient argument is redundant here, you can get the patient from the encounter.

What I was thinking was you were to add 2 separate methods, getOrderGroupsByPatient and getOrderGroupsByEncounter, doesn't make sense to me why you would combine them.

Reply:
Having Patient and Encounter is a hole and recipe for disaster in case some calls the method and patient != encounter.patient